From cf09e2bac1cac560c6019332c37676b919700dda Mon Sep 17 00:00:00 2001
From: Richard Zhao <richard.zhao@freescale.com>
Date: Thu, 19 May 2011 21:59:06 +0800
Subject: [PATCH 976/998] ENGR00143681 GPU: update z430 microcode

The new microcode fix some easy hang issues.

Signed-off-by: Richard Zhao <richard.zhao@freescale.com>
(cherry picked from commit a48455dbea90720cba4be3e827e8b90e6abffd7d)
---
 drivers/mxc/amd-gpu/common/pm4_microcode.inl |   63 +++++++++++++-------------
 1 files changed, 32 insertions(+), 31 deletions(-)

diff --git a/drivers/mxc/amd-gpu/common/pm4_microcode.inl b/drivers/mxc/amd-gpu/common/pm4_microcode.inl
index 03f6f4c..aa7c9fc 100644
--- a/drivers/mxc/amd-gpu/common/pm4_microcode.inl
+++ b/drivers/mxc/amd-gpu/common/pm4_microcode.inl
@@ -29,7 +29,7 @@
 #ifndef PM4_MICROCODE_H
 #define PM4_MICROCODE_H
 
-#define PM4_MICROCODE_VERSION 300684
+#define PM4_MICROCODE_VERSION 322696
 
 #define PM4_MICROCODE_SIZE 768
 
@@ -56,11 +56,11 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x0000ffff, 0xc0284620, 0x000 },
     { 0x00000000, 0xd9004800, 0x000 },
     { 0x00000000, 0x00400000, 0x000 },
-    { 0x00000000, 0x00600000, 0x2a8 },
+    { 0x00000000, 0x00600000, 0x2b0 },
     { 0x00000000, 0xc0200c00, 0x000 },
     { 0x000021fc, 0x0029462c, 0x000 },
     { 0x00000000, 0x00404803, 0x021 },
-    { 0x00000000, 0x00600000, 0x2a8 },
+    { 0x00000000, 0x00600000, 0x2b0 },
     { 0x00000000, 0xc0200000, 0x000 },
     { 0x00000000, 0xc0200c00, 0x000 },
     { 0x000021fc, 0x0029462c, 0x000 },
@@ -78,7 +78,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x0000000e, 0x00404811, 0x000 },
     { 0x00000394, 0x00204411, 0x000 },
     { 0x00000001, 0xc0404811, 0x000 },
-    { 0x00000000, 0x00600000, 0x2a8 },
+    { 0x00000000, 0x00600000, 0x2b0 },
     { 0x000021f9, 0x0029462c, 0x000 },
     { 0x00000008, 0xc0210a20, 0x000 },
     { 0x00000000, 0x14e00000, 0x02d },
@@ -134,7 +134,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000000, 0xc0204800, 0x000 },
     { 0x000021f9, 0x0029462c, 0x000 },
     { 0x00000000, 0x00404802, 0x000 },
-    { 0x0000001f, 0xc0680a20, 0x2a8 },
+    { 0x0000001f, 0xc0680a20, 0x2b0 },
     { 0x000021f9, 0x0029462c, 0x000 },
     { 0x00000000, 0x00404802, 0x000 },
     { 0x8100ffff, 0x00204411, 0x000 },
@@ -142,19 +142,19 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00001fff, 0x40280a20, 0x000 },
     { 0x80000000, 0x40280e20, 0x000 },
     { 0x40000000, 0xc0281220, 0x000 },
-    { 0x00040000, 0x00694622, 0x2b2 },
+    { 0x00040000, 0x00694622, 0x2ba },
     { 0x00000000, 0x00201410, 0x000 },
     { 0x00000000, 0x002f0223, 0x000 },
     { 0x00000000, 0x0ae00000, 0x06d },
     { 0x00000000, 0xc0401800, 0x070 },
     { 0x00001fff, 0xc0281a20, 0x000 },
-    { 0x00040000, 0x00694626, 0x2b2 },
+    { 0x00040000, 0x00694626, 0x2ba },
     { 0x00000000, 0x00201810, 0x000 },
     { 0x00000000, 0x002f0224, 0x000 },
     { 0x00000000, 0x0ae00000, 0x073 },
     { 0x00000000, 0xc0401c00, 0x076 },
     { 0x00001fff, 0xc0281e20, 0x000 },
-    { 0x00040000, 0x00694627, 0x2b2 },
+    { 0x00040000, 0x00694627, 0x2ba },
     { 0x00000000, 0x00201c10, 0x000 },
     { 0x00000000, 0x00204402, 0x000 },
     { 0x00000000, 0x002820c5, 0x000 },
@@ -233,7 +233,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000000, 0x00400000, 0x0a4 },
     { 0x8100ffff, 0x00204411, 0x000 },
     { 0x00000001, 0x00204811, 0x000 },
-    { 0x00040578, 0x00604411, 0x2b2 },
+    { 0x00040578, 0x00604411, 0x2ba },
     { 0x00000000, 0xc0400000, 0x000 },
     { 0x00000000, 0xc0200c00, 0x000 },
     { 0x00000000, 0xc0201000, 0x000 },
@@ -249,7 +249,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000000, 0x0ce00000, 0x0d5 },
     { 0x8100ffff, 0x00204411, 0x000 },
     { 0x00000001, 0x00204811, 0x000 },
-    { 0x00040000, 0x00694624, 0x2b2 },
+    { 0x00040000, 0x00694624, 0x2ba },
     { 0x00000000, 0x00400000, 0x0d6 },
     { 0x00000000, 0x00600000, 0x135 },
     { 0x00000000, 0x002820d0, 0x000 },
@@ -304,7 +304,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000000, 0x17000000, 0x000 },
     { 0x8100ffff, 0x00204411, 0x000 },
     { 0x00000001, 0x00204811, 0x000 },
-    { 0x00040000, 0x00694624, 0x2b2 },
+    { 0x00040000, 0x00694624, 0x2ba },
     { 0x00000000, 0x002820d0, 0x000 },
     { 0x00000000, 0x002f00a8, 0x000 },
     { 0x00000000, 0x0ce00000, 0x000 },
@@ -316,7 +316,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000000, 0x17000000, 0x000 },
     { 0x8100ffff, 0x00204411, 0x000 },
     { 0x00000001, 0x00204811, 0x000 },
-    { 0x00040000, 0x00694624, 0x2b2 },
+    { 0x00040000, 0x00694624, 0x2ba },
     { 0x00000000, 0x002820d0, 0x000 },
     { 0x00000000, 0x002f00a8, 0x000 },
     { 0x00000000, 0x06e00000, 0x000 },
@@ -335,7 +335,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000001, 0x00204811, 0x000 },
     { 0x00000000, 0xc0200800, 0x000 },
     { 0x00007fff, 0x00281a22, 0x000 },
-    { 0x00040000, 0x00694626, 0x2b2 },
+    { 0x00040000, 0x00694626, 0x2ba },
     { 0x00000000, 0x00200c10, 0x000 },
     { 0x00000000, 0xc0201000, 0x000 },
     { 0x80000000, 0x00281a22, 0x000 },
@@ -416,7 +416,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000000, 0xc0200800, 0x000 },
     { 0x00000009, 0x00210222, 0x000 },
     { 0x00000000, 0x14c00000, 0x17d },
-    { 0x00000000, 0x00600000, 0x2af },
+    { 0x00000000, 0x00600000, 0x2b7 },
     { 0x00000000, 0x00200c11, 0x000 },
     { 0x00000016, 0x00203623, 0x000 },
     { 0x00000000, 0x00210222, 0x000 },
@@ -482,7 +482,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x0000000a, 0xc0220e20, 0x000 },
     { 0x00000011, 0x00203623, 0x000 },
     { 0x000021f4, 0x00204411, 0x000 },
-    { 0x0000000a, 0x00614a2c, 0x2af },
+    { 0x0000000a, 0x00614a2c, 0x2b7 },
     { 0x00000005, 0x00210222, 0x000 },
     { 0x00000000, 0x14c00000, 0x1c0 },
     { 0x00000000, 0xc0200000, 0x000 },
@@ -626,7 +626,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00040000, 0x00494624, 0x24c },
     { 0x000021f7, 0x0029122c, 0x000 },
     { 0x00040000, 0x00294624, 0x000 },
-    { 0x00000000, 0x00600000, 0x2b2 },
+    { 0x00000000, 0x00600000, 0x2ba },
     { 0x00000000, 0x002f0222, 0x000 },
     { 0x00000000, 0x0ce00000, 0x252 },
     { 0x00000001, 0x002f0222, 0x000 },
@@ -638,7 +638,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000000, 0x002f0226, 0x000 },
     { 0x00000000, 0x0ae00000, 0x258 },
     { 0x00000000, 0xc0400000, 0x000 },
-    { 0x00040d02, 0x00604411, 0x2b2 },
+    { 0x00040d02, 0x00604411, 0x2ba },
     { 0x00000000, 0x002f0222, 0x000 },
     { 0x00000000, 0x0ae00000, 0x25d },
     { 0x00000010, 0x00211e30, 0x000 },
@@ -662,13 +662,13 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000000, 0xc0204800, 0x000 },
     { 0x00000000, 0x00204806, 0x000 },
     { 0x00005000, 0x00302225, 0x000 },
-    { 0x00040000, 0x00694628, 0x2b2 },
+    { 0x00040000, 0x00694628, 0x2ba },
     { 0x00000001, 0x00302228, 0x000 },
     { 0x00000000, 0x00202810, 0x000 },
-    { 0x00040000, 0x00694628, 0x2b2 },
+    { 0x00040000, 0x00694628, 0x2ba },
     { 0x00000001, 0x00302228, 0x000 },
     { 0x00000000, 0x00200810, 0x000 },
-    { 0x00040000, 0x00694628, 0x2b2 },
+    { 0x00040000, 0x00694628, 0x2ba },
     { 0x00000001, 0x00302228, 0x000 },
     { 0x00000000, 0x00201410, 0x000 },
     { 0x0000060d, 0x00204411, 0x000 },
@@ -707,6 +707,10 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000000, 0x14c00000, 0x29d },
     { 0xa500ffff, 0x00204411, 0x000 },
     { 0x00000001, 0x00404811, 0x299 },
+    { 0x8100ffff, 0x00204411, 0x000 },
+    { 0x00000001, 0x00204811, 0x000 },
+    { 0x00042294, 0x00604411, 0x2ba },
+    { 0x00000000, 0x00200010, 0x000 },
     { 0xa500ffff, 0x00204411, 0x000 },
     { 0x00000000, 0x00204811, 0x000 },
     { 0x000021f4, 0x0029462c, 0x000 },
@@ -717,6 +721,10 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000002, 0x00204811, 0x000 },
     { 0x00000000, 0x00210130, 0x000 },
     { 0xdf7fffff, 0x00283a2e, 0x000 },
+    { 0x8100ffff, 0x00204411, 0x000 },
+    { 0x00000001, 0x00204811, 0x000 },
+    { 0x00042294, 0x00604411, 0x2ba },
+    { 0x00000000, 0x00200010, 0x000 },
     { 0x00000010, 0x0080362a, 0x000 },
     { 0x9700ffff, 0x00204411, 0x000 },
     { 0x00000000, 0x0020480c, 0x000 },
@@ -728,10 +736,10 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000000, 0x00203011, 0x000 },
     { 0x00000010, 0x0080362c, 0x000 },
     { 0x00000000, 0xc0400000, 0x000 },
-    { 0x00000000, 0x1ac00000, 0x2b2 },
+    { 0x00000000, 0x1ac00000, 0x2ba },
     { 0x9f00ffff, 0x00204411, 0x000 },
     { 0xdeadbeef, 0x00204811, 0x000 },
-    { 0x00000000, 0x1ae00000, 0x2b5 },
+    { 0x00000000, 0x1ae00000, 0x2bd },
     { 0x00000000, 0x00800000, 0x000 },
     { 0x00000000, 0x00000000, 0x000 },
     { 0x00000000, 0x00000000, 0x000 },
@@ -770,14 +778,6 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00000000, 0x00000000, 0x000 },
     { 0x00000000, 0x00000000, 0x000 },
     { 0x00000000, 0x00000000, 0x000 },
-    { 0x00000000, 0x00000000, 0x000 },
-    { 0x00000000, 0x00000000, 0x000 },
-    { 0x00000000, 0x00000000, 0x000 },
-    { 0x00000000, 0x00000000, 0x000 },
-    { 0x00000000, 0x00000000, 0x000 },
-    { 0x00000000, 0x00000000, 0x000 },
-    { 0x00000000, 0x00000000, 0x000 },
-    { 0x00000000, 0x00000000, 0x000 },
     { 0x00020143, 0x00020002, 0x000 },
     { 0x00020002, 0x00020002, 0x000 },
     { 0x00020002, 0x00020002, 0x000 },
@@ -805,7 +805,7 @@ uint32 aPM4_Microcode[PM4_MICROCODE_SIZE][3]={
     { 0x00020002, 0x00020002, 0x000 },
     { 0x00020002, 0x00020002, 0x000 },
     { 0x00020002, 0x00020002, 0x000 },
-    { 0x000502b1, 0x00020008, 0x000 },
+    { 0x000502b9, 0x00020008, 0x000 },
 };
 
 #endif
@@ -813,3 +813,4 @@ static const uint32 ME_JUMP_TABLE_START = 740;
 static const uint32 ME_JUMP_TABLE_END   = 768;
 
 #endif
+
-- 
1.7.4.1

